pyblish/util.py

```diff
@@ -26,13 +26,59 @@ def publish(context=None, plugins=None, targets=None):
             defaults to results of discover()
         targets (list, optional): Targets to include for publish session.
 
+    Returns:
+        Context: The context processed by the plugins.
+
     Usage:
         >> context = plugin.Context()
         >> publish(context)     # Pass..
         >> context = publish()  # ..or receive a new
 
     """
 
+    context = api.Context() if context is None else context
+
+    # Since the Context object is a singleton, we can just run though the
+    # publish iterator without assigning any variables.
+    for _ in publish_iter(context, plugins, targets):
+        pass
+
+    return context
+
+
+def publish_iter(context=None, plugins=None, targets=None):
+    """Publish iterator
+
+    This function will process all available plugins of the
+    currently running host, publishing anything picked up
+    during collection.
+
+    Arguments:
+        context (Context, optional): Context, defaults to
+            creating a new context
+        plugins (list, optional): Plug-ins to include,
+            defaults to results of discover()
+        targets (list, optional): Targets to include for publish session.
+
+    Yields:
+        tuple of dict and Context: A tuple is returned with a dictionary and
+            the Context object. The dictionary contains all the result
+            information of a plugin process, and the Context is the Context
+            after the plugin has been processed.
+
+    Usage:
+        >> context = plugin.Context()
+        >> for result, context in util.publish_iter(context):
+               print result
```

Comment:
Could you please make the print-statements Python 3 friendly? `print(result)`

Reply:
And done :)